astropy/nddata/mixins/ndarithmetic.py

```diff
@@ -23,212 +168,638 @@ class NDArithmeticMixin(object):
     When subclassing, be sure to list the superclasses in the correct order
     so that the subclass sees NDData as the main superclass. See
     `~astropy.nddata.NDDataArray` for an example.
+
+    Notes
+    -----
+    This class only aims at covering the most common cases so there are certain
+    restrictions on the saved attributes::
+
+        - ``uncertainty`` : has to be something that has a `NDUncertainty`-like
+          interface for uncertainty propagation
+        - ``mask`` : has to be something that can be used by a bitwise ``or``
+          operation.
+        - ``wcs`` : has to implement a way of comparing with ``=`` to allow
+          the operation.
+
+    But there is a workaround that allows to disable handling a specific
+    attribute and to simply set the results attribute to ``None`` or to
+    copy the existing attribute (and neglecting the other).
+    For example for uncertainties not representing an `NDUncertainty`-like
+    interface you can alter the ``propagate_uncertainties`` parameter in
+    :meth:`NDArithmeticMixin.add`. ``None`` means that the result will have no
+    uncertainty, ``False`` means it takes the uncertainty of the first operand
+    (if this does not exist from the second operand) as the results
+    uncertainty. This behaviour is also explained in the help-page for the
+    different arithmetic operations.
+
+    Notes
+    -----
+    1. It is not tried to decompose the units, mainly due to the internal
+       mechanics of `~astropy.units.Quantity`, so the resulting data might have
+       units like ``km/m`` if you divided for example 100km by 5m. So this
+       Mixin has adopted this behaviour.
+
+    Examples
+    --------
+
+    TODO: Just here because the examples fit more into the main documentation
+    than in here.
+
+    For example::
+
+        >>> from astropy.nddata import *
+        >>> class NDDataWithMath(NDArithmeticMixin, NDData): pass
+        >>> nd = NDDataWithMath([1,2,3], unit='meter')
+        >>> nd_inv = nd.ic_division(1, nd)
```

Comment:
IIRC an issue that comes up is if you try to add an `NDData` to a `Quantity`. Both override `__add__` and `__radd_` and the result ends up being of the type of whatever the first operand is. I think I was the one who raised this in the APE and it could be that I'm wrong...but overriding the arithmetic operators should be a separate PR (if it is ever done).